src/GoTrueClient.ts

```diff
@@ -387,34 +432,40 @@ export default class GoTrueClient {
   }
 
   /**
-   * Returns the user data based on the current session, refreshing the session if necessary.
+   * Gets the current user details if there is an existing session.
    */
-  async getUserFromSession(): Promise<
-    | UserResponse
-    | {
-        data: {
-          user: null
+  async getUser(): Promise<UserResponse> {
+    try {
+      let access_token = ''
+      if (!('Authorization' in this.headers)) {
+        const { session, error } = await this.getSession()
+        if (error) {
+          throw error
         }
```

Comment:
@alaister i'm getting the `Authorization` header from the `this.headers` if there's one - this is to account for the server-side usecase where one would create a new client per request and pass the `Authorization: Bearer token` in the header

Reply:
Sure, makes sense!
Do you think it's cleaner if you do something like `this.headers['Authorization'] ?? session.access_token` instead of having the whole if and extra variable?

Reply:
on hindsight, maybe not because we would avoid invoking `getSession()` for no reason this way